src/Features/CSharp/Portable/Completion/CompletionProviders/PropertySubPatternCompletionProvider.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Completion.Providers;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    internal class PropertySubPatternCompletionProvider : CommonCompletionProvider
+    {
+        public override async Task ProvideCompletionsAsync(CompletionContext context)
+        {
+            var document = context.Document;
+            var position = context.Position;
+            var cancellationToken = context.CancellationToken;
+
+            var workspace = document.Project.Solution.Workspace;
+            var semanticModel = await document.GetSemanticModelForSpanAsync(new TextSpan(position, length: 0), cancellationToken).ConfigureAwait(false);
+            var (type, location) = GetPatternType(document, semanticModel, position, cancellationToken);
+
+            if (type == null)
+            {
+                return;
+            }
+
+            // Find the members that can be tested.
+            IEnumerable<ISymbol> members = semanticModel.LookupSymbols(position, type);
```

Comment:
nit: I would probably consider making these separate variables instead of reassigning, but up to you

Reply:
or just inlining this directly: `semanticModel.LookupSymbols(position, type).Where`

Reply:
nit: Please use 'var'.

Reply:
I get an error in assignment below if I use `var` here. Leaving as-is.

---
In reply to: [185140445](https://github.com/dotnet/roslyn/pull/26474#discussion_r185140445) [](ancestors = 185140445)